.github/workflows/detect-breaking-changes-report.yml

```diff
@@ -116,7 +121,9 @@ jobs:
     # Posts a notification to Slack if a PR has a breaking change and it did not have a breaking change before
     - name: Post to Slack
       id: slack
-      if: steps.levitate-run.outputs.exit_code == 1 && steps.does-label-exist.outputs.result == 0 && steps.levitate-run.outputs.shouldSkip != 'true'
+      env:
+        HAS_SECRETS: ${{ (secrets.SLACK_LEVITATE_WEBHOOK_URL != '') || '' }}
+      if: steps.levitate-run.outputs.exit_code == 1 && steps.does-label-exist.outputs.result == 0 && steps.levitate-run.outputs.shouldSkip != 'true' && env.HAS_SECRETS
```

Comment:
I am not really sure about these checks to be honest. 

If the reporting to Slack or to the Github PR would be optional, then I think this would make a lot of sense, however we would like to be notified about breaking changes, and if the reporting breaks due to a missing permission / github token, then we would like to know about it. If we introduce this change then it will be harder to spot these issues, as the workflow steps will be just skipped instead of failing.

Reply:
If you want it to fail just for the grafana instance, I can add an additional condition. But as long as you *have* a credentials set, you'll still get a failure, it's only if you don't have one set at all that you won't.

I'd find it hard to believe that you wouldn't notice getting zero notifications to Slack if you were still using Slack/notifications. (We use Slack heavily.)

Reply:
We are trying to build tools that we don't need to check constantly to see if they are still working ("there were no notifications since 2 days, can it be that it's broken?"). This workflow is reporting breaking changes and it is not always happening that often that we would immediately see if it goes silent and misses a few PRs. Even if we would notice it after some time, missing a "heads up" for a possible breaking change can cause us head aches.

I have a feeling this workflow does only make sense in the main grafana repo, and don't have any added value for forks, so we could probably introduce a condition (e.g. github.repository == "grafana/grafana") to stop failing on forks?

Reply:
You could certainly do that.

Having just had someone yell at me in another repository for making a PR with "breaking changes" but who couldn't be bothered to explain what the breaking changes were, I'd much rather the CI tell me what the breaking changes are in my fork before I make a PR to upstream.

As such, my personal preference would be to just include that into the HAS_SECRETS condition so that it could be used to send a message to my slack instance if I so chose.

That said, my goal with this PR is to not drown in :x: (see https://github.com/jsoref/grafana/actions) in my forks. I'll do whatever makes this repository happy.